patches/server/0831-Optimize-Util-sequence.patch

```diff
@@ -0,0 +1,42 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: IzzelAliz <csh2001331@126.com>
+Date: Tue, 14 Dec 2021 12:43:03 +0800
+Subject: [PATCH] Optimize Util#sequence
+
+Original method allocates O(n^2) memory on n-size list.
+
+diff --git a/src/main/java/net/minecraft/Util.java b/src/main/java/net/minecraft/Util.java
+index f6561599391583ba7d669af42b5716cda0df2d68..15b6e36242fd91c66bc15f7407c22b6d78a24457 100644
+--- a/src/main/java/net/minecraft/Util.java
++++ b/src/main/java/net/minecraft/Util.java
+@@ -391,21 +391,16 @@ public class Util {
+     }
+ 
+     public static <V> CompletableFuture<List<V>> sequence(List<? extends CompletableFuture<? extends V>> futures) {
+-        return futures.stream().reduce(CompletableFuture.completedFuture(Lists.newArrayList()), (completableFuture, completableFuture2) -> {
+-            return completableFuture2.thenCombine(completableFuture, (object, list) -> {
+-                List<V> list2 = Lists.newArrayListWithCapacity(list.size() + 1);
+-                list2.addAll(list);
+-                list2.add(object);
+-                return list2;
+-            });
+-        }, (completableFuture, completableFuture2) -> {
+-            return completableFuture.thenCombine(completableFuture2, (list, list2) -> {
+-                List<V> list3 = Lists.newArrayListWithCapacity(list.size() + list2.size());
+-                list3.addAll(list);
+-                list3.addAll(list2);
+-                return list3;
++        // Paper start - optimize
++        return CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]))
++            .thenApply(v -> {
++                var list = Lists.<V>newArrayListWithCapacity(futures.size());
```

Comment:
Var makes this more awkward to read, just do `List<V> list = new ArrayList<>(futures.size());`